default.py

```diff
@@ -47,6 +49,14 @@ def addon_log(string):
     msg = '%s: %s' % (LOGGING_PREFIX, string)
     xbmc.log(msg=msg, level=xbmc.LOGDEBUG)
 
+def ShowBusyDialog():
+    busydialog.create()
+
+def HideBusyDialog():
+    try:
+        busydialog.close()
+    except RuntimeError,e:
```

Comment:
Yes I did encounter errors. Pretty long explanation, but here goes:
There are more calls to closing the busy dialog than there are flows that open one. When you call close() when the dialog is not open, it throws an error. Unfortunately it appears you cannot ask the busydialog if it is currently open (or I haven't found it). This was one way to work around that in a fail safe way.

Background:
I also thought of maintaining a 'busydialogisopen' boolean state in the script. But the close()-call that actually triggered the runtime error is in the entrypoint of the script itself, when we can't possibly have opened a dialog yet. Because that fact is so obvious, I thought you put it there intentionally; perhaps it is closing a possible busy dialog from outside the addon / something in Kodi itself? Maintaining our own  'busydialogisopen' boolean state would make that dialog stay open because we haven't opened it ourselves.